### PR TITLE
Restructure repository + improve documentation

### PR DESCRIPTION
**Pull request by sgraband**
Overall:
- changed the example from EAM to the Coffee model
- changed from left/right/origin to source/target/base
- formatted the code
- Addded READMEs (Documentation) for all parts.

emf-compare-adaptor:
- Added the required libs to the repository
- The model is now passed via an CLI arg
- Removed the old model
- Added DiffObject and DiffObjectManagers
- Separated the graphicalID from the UUID for diffs
- Restructured the code to simplify it
- Added a launch config to debug the jar
- Put all icons in one class
- improve error handling

client:
- Restructured the project to contain only the extension and examples
- Added linitng and formatting
- Changed the building workflow
- The server configuration is now done by the example
- Other minor improvements.

### CODE REVIEW

File: client/comparison-extension/package.json [hunk 1]
```diff
@@ -10,7 +10,6 @@
     "style"
   ],
   "dependencies": {
-    "@eclipse-emfcloud/modelserver-theia": "next",
     "@fortawesome/fontawesome-free": "^5.14.0",
```
Comment (eneufeld) [inline client/comparison-extension/package.json:13]: I think it is more a follow up, but we should use the codicon icons to be more theia compatible.

File: client/comparison-extension/package.json [hunk 2]
```diff
@@ -10,7 +10,6 @@
...
     "@jsonforms/core": "2.4.0",
```
Comment (eneufeld) [inline client/comparison-extension/package.json:14]: Also update this to 3.0.0 at some point

File: client/comparison-extension/src/browser/graphical/graphical-comparison-contribution.ts [hunk 3]
```diff
@@ -13,34 +13,40 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
-import { CommandRegistry, Command, MenuModelRegistry, SelectionService } from '@theia/core/lib/common';
+/* eslint-disable brace-style */
```
Comment (eneufeld) [inline client/comparison-extension/src/browser/graphical/graphical-comparison-contribution.ts]: why do we need this?

File: client/comparison-extension/src/browser/graphical/graphical-comparison-extension-frontend-module.ts [hunk 4]
```diff
@@ -13,19 +13,18 @@
...
-import '../../../style/elements.css';
-
-import { ContainerModule } from 'inversify';
 import { bindViewContribution, WidgetFactory } from '@theia/core/lib/browser';
-import { GraphicalComparisonWidget, GraphicalComparisonWidgetOptions } from './graphical-comparison-widget';
-import { GraphicalComparisonContribution } from './graphical-comparison-contribution';
 import { TabBarToolbarContribution } from '@theia/core/lib/browser/shell/tab-bar-toolbar';
+import { ContainerModule } from 'inversify';
+import '../../../style/elements.css';
+import '../../../style/index.css';
```
Comment (eneufeld) [inline client/comparison-extension/src/browser/graphical/graphical-comparison-extension-frontend-module.ts:20]: I assume the order matters here for the css? We should check

**Review by eneufeld — APPROVED** [review]
Do we need to check in client/examples/coffee-example/coffee-comparison-extension/model/org.eclipse.emfcloud.coffee.model-0.1.0-SNAPSHOT.jar ?
Can't we consume the libs (emf-compare-adaptor/lib) using a pom or a target?
Looks ok overall.
There are some commented out lines which could be cleaned up.

File: client/comparison-extension/package.json (hunk 1 above)
Comment (sgraband) [reply client/comparison-extension/package.json:13]: I will add it to the list and create an issue for this.

File: client/comparison-extension/package.json (hunk 2 above)
Comment (sgraband) [reply client/comparison-extension/package.json:14]: I will add it to the list and create an issue for this.

File: client/comparison-extension/src/browser/graphical/graphical-comparison-extension-frontend-module.ts (hunk 4 above)
Comment (sgraband) [reply client/comparison-extension/src/browser/graphical/graphical-comparison-extension-frontend-module.ts:20]: I checked the css files. The contents are not overlapping, so the order should not matter.

**Comment by sgraband** [comment]
The coffee-model snapshot jar is part of the examples. We could download it from somewhere, but i do not think it is worth the effort right now.
Regarding the libs: As far as i can tell they are not available via maven. We could make it a p2 project, but this will be a bit of work, as we currently have no p2 build.

**Comment by eneufeld** [comment]
I see, hmm we should add the dependencies on our issue list.
Can be merged otherwise. 
Thank you!

**Pull request merged**